src/include/cell.i

```diff
@@ -193,14 +190,42 @@ __wt_timestamp_value_check(
  */
 static inline void
 __cell_pack_timestamp_value(WT_SESSION_IMPL *session,
-    uint8_t **pp, wt_timestamp_t start_ts, wt_timestamp_t stop_ts)
+    uint8_t **pp, wt_timestamp_t *start_tsp, wt_timestamp_t *stop_tsp)
 {
-	__wt_timestamp_value_check(session, start_ts, stop_ts);
+	wt_timestamp_t start_ts, stop_ts;
 
-	if (__wt_process.page_version_ts) {
-		/* Start timestamp, stop timestamp difference. */
-		(void)__wt_vpack_uint(pp, 0, start_ts);
-		(void)__wt_vpack_uint(pp, 0, stop_ts - start_ts);
+	__wt_timestamp_value_check(session, *start_tsp, *stop_tsp);
+
+	/*
+	 * Finalize the timestamps.
+	 *
+	 * TIMESTAMP-FIXME
+	 * Values (presumably) have associated transaction IDs, but we haven't
+	 * yet decided how to handle them. For now, ignore them in determining
+	 * value durability.
+	 */
+	if (*stop_tsp != WT_TS_MAX &&
+	    __wt_txn_visible_all(session, WT_TXN_NONE, *stop_tsp)) {
+		*start_tsp = WT_TS_NONE;
+		*stop_tsp = WT_TS_MAX;
```

Comment:
If this test were true, it would mean that the value was removed, and the remove is now stable.  In that case, unless the oldest timestamp has been updated while reconciliation is running, we shouldn't be storing the record at all (i.e., it should have been filtered out by `__rec_upd_select`).

For now, I think this check should be flipped to consider start timestamps for the common case where an insert becomes stable and isn't removed.

Reply:
I don't completely follow this comment -- please feel free to update the code to match what you expect.

The test I'm trying to make here is if the specified timestamp pair is globally visible. There are paths through here that aren't removes, or even through `__rec_upd_select()`, and as I think you're pointing out, the transaction state may have changed since timestamps for this value were chosen. In other words, I'm thinking of this as "we chose a value to write and it has some timestamps, but before we write it, check to see if those timestamps are necessary based on the current state of the system".

Reply:
I pushed the checks I expected in a055e22: please take a look and let me know if that doesn't make sense.